src/gt4py/analysis/passes.py

```diff
@@ -1201,27 +1201,27 @@ def apply(cls, node: gt_ir.StencilImplementation) -> Set[str]:
 
         def __call__(self, node: gt_ir.StencilImplementation) -> Set[str]:
             assert isinstance(node, gt_ir.StencilImplementation)
-            self.demotables: Dict[str, str] = {
-                field_name: "" for field_name in node.temporary_fields
+            """Dictionary mapping temporaries to their most recently referenced stage."""
+            self.demotables: Dict[str, Optional[str]] = {
+                temp_field: None for temp_field in node.temporary_fields
             }
```

Comment:
I believe docstrings should go after the variable definition. See https://www.python.org/dev/peps/pep-0224/#id4